### ISSUE TITLE
Verifying iptables firewall config.

### ISSUE THREAD

**Issue by GeneralZiggy** [pivpn/pivpn#255]
Hi All,

My VPN is working fine, I'd just like to verify if my firewall config is secure with y'all who know more than I do.  Any help is much appreciated!

**Commands:**
Entered the commands on this site > [https://arashmilani.com/post?id=53](url)
iptables-save > /etc/iptables/rules.v4  as root

`
**admin-pi@pi-ovpn:~ $ sudo iptables -L**
Chain INPUT (policy ACCEPT)
target     prot opt source               destination
ACCEPT     udp  --  anywhere             anywhere             state NEW udp dpt:openvpn
ACCEPT     all  --  anywhere             anywhere

Chain FORWARD (policy ACCEPT)
target     prot opt source               destination
ACCEPT     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere             state RELATED,ESTABLISHED
ACCEPT     all  --  anywhere             anywhere             state RELATED,ESTABLISHED

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination
ACCEPT     all  --  anywhere             anywhere
`

` 
Generated by iptables-save v1.4.21 on Tue May  9 23:44:28 2017
*filter
:INPUT ACCEPT [796:66773]
:FORWARD ACCEPT [0:0]
:OUTPUT ACCEPT [589:73244]
-A INPUT -i eth0 -p udp -m state --state NEW -m udp --dport 1194 -j ACCEPT
-A INPUT -i tun+ -j ACCEPT
-A FORWARD -i tun+ -j ACCEPT
-A FORWARD -i tun+ -o eth0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -i eth0 -o tun+ -m state --state RELATED,ESTABLISHED -j ACCEPT
-A OUTPUT -o tun+ -j ACCEPT
COMMIT
# Completed on Tue May  9 23:44:28 2017
# Generated by iptables-save v1.4.21 on Tue May  9 23:44:28 2017
*nat
:PREROUTING ACCEPT [34:2384]
:INPUT ACCEPT [34:2384]
:OUTPUT ACCEPT [32:2591]
:POSTROUTING ACCEPT [32:2591]
`

**Comment by 0-kaladin** [comment]
Looks wide open.  did you intend to add rules?

**Comment by GeneralZiggy** [comment]
Yes that's what I thought too. I followed the [Raspberry Pi - OpenVPN Setup via PiVPN](https://www.youtube.com/watch?v=pUBMcsvJfe4) tutorial on the PiVPN site. Everything in the tutorial is crystal clear except for the iptables configuration.

I think I am mostly unsure about seeing "ACCEPT all -- anywhere anywhere" under the INPUT chain which I am thinking would mean that my pi would be accepting un-established connections.
I ran "iptables -vL" and saw that this was only for the established "tun+" interface which mostly alleviated that uncertainty.

I know I should do more reading up on iptables, but my question basically boils down to this: I already have firewall rules set up on my router. So do I even really need any additional rules set up on the pi for a secure vpn configuration?

**Comment by 0-kaladin** [comment]
No not really.  your pi should be on your internal network completely not exposed to the internet.  and then in your router you will port forward ONLY 1194 (or your chosen vpn port) to the pi.  So unless you want to further configure iptables to lock down the pi from devices on your internal network, you do not need to do anything else to secure it from the internet side.

**Comment by cfcolaco** [comment]
@GeneralZiggy Actually you should be safe since you probably have the most common setup:
Router ( Firewall ) || Local Network 

sho theoretically you should be safe, however, if you want to add extra safety you can change  Input and forward to DROP
making something like Router (Firewall) || local network || Pi Firewall || Pi system

:INPUT DROP [796:66773]
:FORWARD DROP [0:0]
:OUTPUT ACCEPT [589:73244]

which will be "overridden" by the specific rules you add 

-A INPUT -i eth0 -p udp -m state --state NEW -m udp --dport 1194 -j ACCEPT
-A INPUT -i tun+ -j ACCEPT
-A FORWARD -i tun+ -j ACCEPT
-A FORWARD -i tun+ -o eth0 -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -i eth0 -o tun+ -m state --state RELATED,ESTABLISHED -j ACCEPT
-A OUTPUT -o tun+ -j ACCEPT

In other words ... Firewall will DROP every packet except those you say it should ACCEPT


Hope this helps you out! ;)

**Comment by cfcolaco** [comment]
@GeneralZiggy did you got your answer here? anything else you'd like to discuss? please close the issue if you're done!

**Comment by cfcolaco** [comment]
@redfast00 think this one can be closed also!
